zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/DynamicSecurityFileLoader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal.elasticsearch;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Properties;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.PASSWORD_PROP;
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.USERNAME_PROP;
+
+/**
+ * Load username/password from security file.
+ */
+class DynamicSecurityFileLoader {
+  static final String SECURITY_FILE_REFRESH_INTERVAL_IN_SECOND =
+    "zipkin.storage.elasticsearch.security-file-refresh-interval-in-second";
+
+  private final String securityFilePath;
+
+  private final BasicCredentials basicCredentials;
+
+  public DynamicSecurityFileLoader(BasicCredentials basicCredentials, String securityFilePath) {
+    this.basicCredentials = basicCredentials;
+    this.securityFilePath = securityFilePath;
+  }
+
+  @Scheduled(fixedRateString = "${" + SECURITY_FILE_REFRESH_INTERVAL_IN_SECOND +"}000")
```

Comment:
maybe an expression `* 1000` here does that bring any extra safety ? Can the user specify `10.5` as value, and if yes what would happen ?

Reply:
It will cause a "NumberFormatException" because the type of this param is `Integer`. I have added `*1000` to the setter of it.